### PR TITLE
[WIP] Add possibility to inject readinessGates via Sidecar Injector

### PR DESCRIPTION
**Pull request by kuzaxak**
https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate

#19757

### CODE REVIEW

**Comment by howardjohn** [comment]
just want to clarify - if you modified the injection template today and
added readiness gates it wouldn't be added properly without this change?

**Comment by kuzaxak** [comment]
>  if you modified the injection template today and added readiness gates it wouldn't be added properly without this change?

Yes, because injector ignores unknown fields.

**Comment by kuzaxak** [comment]
@howardjohn Hi, I tested it on my local cluster. Everything looks fine. 

Can I remove WIP?

File: pkg/kube/inject/testdata/inject/auth.yaml.injected [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
     metadata:
       annotations:
         sidecar.istio.io/interceptionMode: REDIRECT
-        sidecar.istio.io/status: '{"version":"","initContainers":["istio-init"],"containers":["istio-proxy"],"volumes":["istio-envoy","istio-certs"],"imagePullSecrets":null}'
+        sidecar.istio.io/status: '{"version":"","initContainers":["istio-init"],"containers":["istio-proxy"],"volumes":["istio-envoy","istio-certs"],"imagePullSecrets":null,"readinessGates":null}'
```
Comment (howardjohn) [inline pkg/kube/inject/testdata/inject/auth.yaml.injected]: can you add a test under inject/ as well? otherwise lgtm

Comment (kuzaxak) [reply pkg/kube/inject/testdata/inject/auth.yaml.injected]: Okay, I will try to add injecting tests.

File: istioctl/cmd/testdata/deployment/hello.yaml.injected [hunk 2]
```diff
@@ -14,7 +14,7 @@ spec:
   template:
     metadata:
       annotations:
-        sidecar.istio.io/status: '{"version":"2343d4598565fd00d328a3388421ee637d25d3f7068e7d5cadef374ee1a06b37","initContainers":["istio-init"],"containers":["istio-proxy"],"volumes":null,"imagePullSecrets":null}'
+        sidecar.istio.io/status: '{"version":"2343d4598565fd00d328a3388421ee637d25d3f7068e7d5cadef374ee1a06b37","initContainers":["istio-init"],"containers":["istio-proxy"],"volumes":null,"imagePullSecrets":null,"readinessGates":null}'
```
Comment (hzxuzhonghu) [inline istioctl/cmd/testdata/deployment/hello.yaml.injected]: looks good, can we remove the null fields?

**Comment by howardjohn** [comment]
Thanks for the PR @kuzaxak, sorry this fell through the cracks a bit. if you rebase and mention me I can get this approved for you, LGTM

**Pull request closed without merging**